src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvThread.cs

```diff
@@ -302,7 +300,7 @@ private void OnPost()
                 wasWork = DoPostWork();
                 wasWork = DoPostCloseHandle() || wasWork;
                 loopsRemaining--;
-            } while (wasWork && loopsRemaining > 0);
+            } while (!_stopping && wasWork && loopsRemaining > 0);
```

Comment:
Today we do at least 8 loops after work has been posted to check for more work (on the libuv thread). When shutting down, this flag just ensures that we don't waste any time looking for more work.